### ISSUE TITLE
Clipboard.setData crashes the app

### ISSUE THREAD

**Issue by kirill-21** [flutter/flutter#121976]
```dart
String? key

void main() {
  try {
    Clipboard.setData(ClipboardData(text: key));
  } catch (_) {
    /* Do nothing */
  }
}
```

ClipboardData accepts nullable strings, however, such setter crashes the app.

**Platform: Windows. Flutter: 3.7.6 from stable channel**

**Comment by danagbemava-nc** [comment]
Issue is reproducible on the latest versions of flutter. 

There are no logs when the app crashes. It just logs `Lost connection to device`.

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() {
  runApp(
    MaterialApp(
      home: Scaffold(
        body: Column(
          children: [
            ListTile(
              title: const Text('ListTile'),
              subtitle: const Text('forbidden, compact'),
              onTap: () {
                try {
                  Clipboard.setData(ClipboardData(text: null));
                } catch (_) {
                  /* Do nothing */
                }
              },
            ),
          ],
        ),
      ),
    ),
  );
}
```
</details>

<details>
<summary>flutter doctor -v</summary>


```
[√] Flutter (Channel stable, 3.7.6, on Microsoft Windows [Version 10.0.19045.2486], locale en-GH)
    • Flutter version 3.7.6 on channel stable at C:\src\sdks\stable\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 12cb4eb7a0 (5 days ago), 2023-03-01 10:29:26 -0800
    • Engine revision ada363ee93
    • Dart version 2.19.3
    • DevTools version 2.20.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at C:\Users\Deana\AppData\Local\Android\sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.2.32526.322
    • Windows 10 SDK version 10.0.22000.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] IntelliJ IDEA Community Edition (version 2022.1)
    • IntelliJ at C:\Users\Deana\AppData\Local\JetBrains\Toolbox\apps\IDEA-C\ch-0\221.5921.22
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] IntelliJ IDEA Ultimate Edition (version 2022.1)
    • IntelliJ at C:\Users\Deana\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\221.5921.22
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code (version 1.76.0)
    • VS Code at C:\Users\Deana\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.60.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19045.2486]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 110.0.5481.178
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 110.0.1587.63

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[!] Flutter (Channel master, 3.9.0-1.0.pre.50, on Microsoft Windows [Version 10.0.19045.2486], locale en-GH)
    • Flutter version 3.9.0-1.0.pre.50 on channel master at c:\src\sdks\master\flutter
    ! Warning: flutter on your path resolves to C:\src\sdks\stable\flutter\bin\flutter, which is not inside your
      current Flutter SDK checkout at c:\src\sdks\master\flutter. Consider adding c:\src\sdks\master\flutter\bin to the
      front of your path.
    ! Warning: dart on your path resolves to C:\src\sdks\stable\flutter\bin\dart, which is not inside your current
      Flutter SDK checkout at c:\src\sdks\master\flutter. Consider adding c:\src\sdks\master\flutter\bin to the front of
      your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6afaeab23a (4 hours ago), 2023-03-06 01:42:24 -0500
    • Engine revision b4297f3cc8
    • Dart version 3.0.0 (build 3.0.0-290.0.dev)
    • DevTools version 2.22.2
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly
      to perform update checks and upgrades.

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at C:\Users\Deana\AppData\Local\Android\sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.2.32526.322
    • Windows 10 SDK version 10.0.22000.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] IntelliJ IDEA Community Edition (version 2022.1)
    • IntelliJ at C:\Users\Deana\AppData\Local\JetBrains\Toolbox\apps\IDEA-C\ch-0\221.5921.22
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] IntelliJ IDEA Ultimate Edition (version 2022.1)
    • IntelliJ at C:\Users\Deana\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\221.5921.22
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code (version 1.76.0)
    • VS Code at C:\Users\Deana\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.60.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19045.2486]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 110.0.5481.178
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 110.0.1587.63

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

**Comment by loic-sharma** [comment]
Hello! Thank you for reporting this. It looks like making the `ClipboardData.text` nullable was a mistake during Flutter's null migration.

Currently the different platforms behave unexpectedly if you set your clipboard text to `null`:

1. [iOS will set your clipboard content to the text `"null"`](https://github.com/flutter/engine/blob/12f2fdff743a56ced0557c042c29d617af398873/shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm#L276).
2. [Web assumes incorrectly that the text is non-null](https://github.com/flutter/engine/blob/12f2fdff743a56ced0557c042c29d617af398873/lib/web_ui/lib/src/engine/clipboard.dart#L128)
3. [Linux will return an error if the text is null](https://github.com/flutter/engine/blob/12f2fdff743a56ced0557c042c29d617af398873/shell/platform/linux/fl_platform_plugin.cc#L104-L105)
4. Windows crashes

In other words, customers who set their clipboard to `null` today are broken. I will update the framework to disallow setting the clipboard to `null`, and, make the Windows embedder not crash for unexpected clipboard messages.
